iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveIcebergRESTCatalogClientAdapter.java

```diff
@@ -0,0 +1,2426 @@
+  public Database getDatabase(String databaseName) throws NoSuchObjectException, MetaException, TException {
+    return restCatalog.listNamespaces(Namespace.empty()).stream()
+        .filter(namespace -> namespace.levels()[0].equals(databaseName)).map(namespace -> {
+          Database database = new Database();
+          database.setName(String.join(NAMESPACE_SEPARATOR, namespace.levels()));
+          Map<String, String> namespaceMetadata = restCatalog.loadNamespaceMetadata(Namespace.of(databaseName));
+          database.setLocationUri(namespaceMetadata.get(LOCATION));
+          database.setCatalogName("REST");
+          database.setOwnerName(namespaceMetadata.get(DB_OWNER));
+          try {
+            database.setOwnerType(PrincipalType.valueOf(namespaceMetadata.get(DB_OWNER_TYPE)));
+          } catch (Exception e) {
+            LOG.warn("Can not set ownerType: {}", namespaceMetadata.get(DB_OWNER_TYPE), e);
+          }
+          return database;
+        }).findFirst().get();
+  }
+
+  @Override
+  public Database getDatabase(String catalogName, String databaseName)
+      throws NoSuchObjectException, MetaException, TException {
+    return getDatabase(databaseName);
+  }
+
+  @Override
+  public Table getTable(String dbName, String tableName) throws MetaException, TException, NoSuchObjectException {
+    org.apache.iceberg.Table icebergTable = null;
+    try {
+      icebergTable = restCatalog.loadTable(TableIdentifier.of(dbName, tableName));
+    } catch (NoSuchTableException exception) {
+      throw  new NoSuchObjectException();
+    }
+    Table hiveTable = convertIcebergTableToHiveTable(icebergTable);
+    return hiveTable;
+  }
+  private Table convertIcebergTableToHiveTable(org.apache.iceberg.Table icebergTable) {
+    Table hiveTable = new Table();
+    TableMetadata metadata = ((BaseTable) icebergTable).operations().current();
+    HMSTablePropertyHelper.updateHmsTableForIcebergTable(metadata.metadataFileLocation(), hiveTable,
+        metadata, null, true, maxHiveTablePropertySize, null);
+    hiveTable.getParameters().put(CATALOG_NAME, CatalogUtil.ICEBERG_CATALOG_TYPE_REST);
+    hiveTable.setTableName(getTableName(icebergTable));
+    hiveTable.setDbName(getDbName(icebergTable));
+    StorageDescriptor storageDescriptor = new StorageDescriptor();
+    hiveTable.setSd(storageDescriptor);
+    hiveTable.setTableType("EXTERNAL_TABLE");
+    hiveTable.setPartitionKeys(new LinkedList<>());
+    List<FieldSchema> cols = new LinkedList<>();
+    storageDescriptor.setCols(cols);
+    storageDescriptor.setLocation(icebergTable.location());
+    storageDescriptor.setInputFormat(DEFAULT_INPUT_FORMAT_CLASS);
+    storageDescriptor.setOutputFormat(DEFAULT_OUTPUT_FORMAT_CLASS);
+    storageDescriptor.setBucketCols(new LinkedList<>());
+    storageDescriptor.setSortCols(new LinkedList<>());
+    storageDescriptor.setParameters(Maps.newHashMap());
+    if (icebergTable.properties().containsKey(SERDE_INFO)) {
+      // TODO
```

Comment:
is this TODO to be done later in this PR? Else, can you please add a explanation why it needs to be done in another PR.

Reply:
Fixed.